internal/controllers/clusterrequest_controller.go

```diff
@@ -1730,8 +1823,80 @@ func (r *ClusterRequestReconciler) findManagedClusterForClusterRequest(
 	}
 }
 
+// findPoliciesForClusterRequestsForDelete: Finds if the deleted child policy matches managed clusters
+// created through ClusterRequests.
+func (r *ClusterRequestReconciler) findPoliciesForClusterRequestsForDelete(
+	ctx context.Context, e event.DeleteEvent, q workqueue.RateLimitingInterface) {
+
+	policy := e.Object.(*policiesv1.Policy)
+	listOpts := []client.ListOption{
+		client.MatchingFields{
+			"status.clusterInstanceRef.name": policy.Namespace,
+		},
+	}
+
+	clusterRequests := &oranv1alpha1.ClusterRequestList{}
+	err := r.Client.List(ctx, clusterRequests, listOpts...)
+	if err != nil {
+		r.Logger.Error("[findPolicyForManagedClustersOfClusterRequests] Error listing ClusterRequests. ", "Error: ", err)
+		return
+	}
+
+	for _, clusterRequest := range clusterRequests.Items {
+		q.Add(reconcile.Request{NamespacedName: types.NamespacedName{
+			Name:      clusterRequest.Name,
+			Namespace: clusterRequest.Namespace,
+		}})
+	}
+}
+
+// findPoliciesForClusterRequestsForUpdate: Finds if the child policy matches managed clusters
+// created through ClusterRequests.
+func (r *ClusterRequestReconciler) findPoliciesForClusterRequestsForUpdate(
+	ctx context.Context, e event.UpdateEvent, q workqueue.RateLimitingInterface) {
+
+	oldPolicy := e.ObjectOld.(*policiesv1.Policy)
+
+	// Get the clusterRequests that created a managed cluster with the same name
+	// as the policy namespace.
+	listOpts := []client.ListOption{
+		client.MatchingFields{
+			"status.clusterInstanceRef.name": oldPolicy.Namespace,
+		},
+	}
```

Comment:
The Clusterrequest labels are added to the generated ClusterInstance. What about querying the clusterInstance instead? See `findManagedClusterForClusterRequest`

Reply:
That would work equally well, I think.
I made the update since it might be a slightly clearer query.